src/components/fileviewer.js

```diff
@@ -65,16 +54,43 @@ export default class FileViewerContainer extends Component {
       },
       limit: 1000,
       format: 'list',
-    })
-      .then((data) => {
-        this.parseCoverage(data.data);
-      });
+    };
+  }
+
+  async componentDidMount() {
+    await this.fetchData();
   }
 
   setSelectedLine(selectedLineNumber) {
     this.setState({ selectedLine: selectedLineNumber });
   }
 
+  async fetchData() {
+    /* Fetch source code from hg */
+    try {
+      const text = await FetchAPI.getRawFile(this.revision, this.path);
+      this.setState({ parsedFile: text.split('\n'), srcFetchStatus: true });
+    } catch (error) {
+      console.error(error);
+      this.setState({
+        appError: 'We did not manage to fetch source file from hg.mozilla',
+        srcFetchStatus: false,
+      });
+    }
+    /* Fetch coverages from ActiveData */
```

Comment:
I am not sure if performing both fetches in one function is a good idea: Fetching is describing *how* the code works, rather than *what* it provides.  Having a named function responsible for getting source code, and named function for getting (and parsing) coverage is more descriptive. It allows you to perform one, without the other (if there is ever a need) plus we can call them in parallel so the network requests are done together, rather than waiting on one.